### PR TITLE
module: fix bug with submodule import

### LINKED ISSUE #10211: Submodule imports may be broken
**V version: 0.2.2 d8d05e0** 
**OS: Windows 10, fresh install, not using WSL** 

**What did you do?**
I have the following directory structure:
```
.
|--- mymodule
|    |--- main_functions.v
|    '--- submodule
|          '--- sub_functions.v
'--- main.v
```
Inside `main_functions.v` I have:
```
module mymodule
pub fn add_xy(x int, y int) int {
    return x + y
}
```
And inside `sub_functions.v` I have:
```
module submodule
pub fn sub_xy(x int, y int) int {
    return x - y
}
```
Finally, in `main.v` I have:
```
import mymodule { add_xy }
import mymodule.submodule { sub_xy }

fn main() {
    println(add_xy(2, 3)) // expected: 5
    println(sub_xy(10, 7)) // expected: 3
}
```

**What did you expect to see?**
I expected my main.v file to import the function `add_xy` from the module `mymodule`. Then, it should import `sub_xy` from the module `submodule` located inside the module `mymodule`. Finally, it should run these functions with the parameters shown in `main.v` and print the commented expectations.

**What did you see instead?**
I got the following error:
```
builder error: bad module definition: .\v imports module "mymodule.submodule" but C:\Users\lucas\Documents\VProject\mymodule\submodule\sub_functions.v is defined as module `submodule`
    2 | import mymodule.submodule { sub_xy } (this is in main.v)
```

What could be done to fix the issue? Is this really a bug or just my mistake?

### PR DESCRIPTION
**Pull request by Zymlex**
fixed full module name (for `ast.Module.name` [#L2600](https://github.com/vlang/v/blob/d39a55ac11861b2b034f971edfab4708c6b6d01f/vlib/v/parser/parser.v#L2600))

fixed path separator

fix #10211

### CODE REVIEW

**Comment by joe-conigliaro** [comment]
Hi @Zymlex , thanks for the code there is an issue I can see however (unless some code was left out)

This will only work in case where an absolute path was passed to v (pref.path), this can be fixed to work when relative path is passed by converting pref.path to abs_path, using os.real_path on it.

I tested on my local machine with, and was receiving the same error. I am going to add some suggested code, I'm just testing something currently.

**Comment by Larpon** [comment]
... also adding the examples as regression tests would be nice before merging

**Comment by joe-conigliaro** [comment]
I have supplied a modified version of just the `mod_path_to_full_name` function here: https://gist.github.com/joe-conigliaro/2424047643d230ed7f92f807a647636f

This works for the case described and building v, I haven't tried tests yet. But something along these lines should work

**Comment by joe-conigliaro** [comment]
Also, in the gist I provides, it will now bypass the v.mod checking in relative cases, so it might be best to move the code from the first if branch, to inside the else, below where v.mod checking code, so it only runs when the v.mod checks fail?

File: vlib/v/util/module.v [hunk 1]
```diff
@@ -110,8 +110,9 @@ pub fn mod_path_to_full_name(pref &pref.Preferences, mod string, path string) ?s
 			}
 		}
 	}
-	if os.is_abs_path(pref.path) && os.is_abs_path(path) && os.is_dir(path)/* && path.contains(mod)*/ {
-		rel_mod_path := path.replace(pref.path.all_before_last(os.path_separator) + os.path_separator, '')
+	if os.is_abs_path(pref.path) && os.is_abs_path(path) && os.is_dir(path) //&& path.contains(mod) {
```
Comment (spytheman) [inline vlib/v/util/module.v:113]: vfmt did an oopsie :-| ... it has trouble handling inline comments inside complex expressions

**Comment by spytheman** [comment]
> ... also adding the examples as regression tests would be nice before merging

I second that - we do need as many regression tests for the different module importing edge cases as possible.

File: vlib/v/util/module.v (hunk 1 above)
Comment (JalonSolov) [reply vlib/v/util/module.v]: inline comments are ebil... I thought we seriously considered removing support for them.

**Comment by Zymlex** [comment]
@joe-conigliaro Thanks for the samples, I try to use them in the fix, unfortunately, in the proposed form, they lead to crashes V (with fix) during V self.

**Comment by joe-conigliaro** [comment]
@Zymlex no problem, that's a bit strange I ran CI with the second version and it passed. hmm

**Comment by Zymlex** [comment]
> are you on windows?

Yes.
